tools/manifest/sourcefile.py

```diff
@@ -308,11 +308,7 @@ def hash(self):
                 content = f.read()
 
             data = b"".join((b"blob ", b"%d" % len(content), b"\0", content))
-            hash_str = hashlib.sha1(data).hexdigest()  # type: str
-            if PY3:
-                self._hash = hash_str.encode("ascii")
-            else:
-                self._hash = hash_str
+            self._hash = hashlib.sha1(data).hexdigest()
```

Comment:
So this ends up as different types in Py2 and 3 which I think we want to avoid.

Reply:
Hmm I don't get this, `hexdigest()` generates <str> in py2 and py3 AFAIK

Reply:
Right, but `str` is binary in Python 2 and text in Python 3, so different types with the same name.

Reply:
Note that we're already returning str in many properties of this same class, but I get what you mean.

I'll try to find the right spot to decode it when converting the json then...

Reply:
Right, this setup where we are storing bytes in Python 2 but binary in Python 3 worries me; I feel like it's going to end up causing problems down the line. I'd really much prefer we make data in the program one or the other and do conversions only when we access external functions which accept different types in Python 2 and 3.

Reply:
Thing is, if we store the hash in bytes, then just before converting the manifest data to json, we'd need to transverse the whole dictionary and replace the binaries with strings to allow the json module to process it.

So maybe another option is to call `unicode()` for python2 and store unicode data?